### PR TITLE
Issue 532: 373/369 fix

### PR DESCRIPTION
**Pull request by christinem**
This fixes Issue #532.

This PR is built on #531.

### CODE REVIEW

**Comment by david-yz-liu** [comment]
Seems like it would be better to add an extra `name` field to the post objects, rather than pass in an extra parameter.

**Comment by christinem** [comment]
@david-yz-liu that thought crossed my mind as well, actually. I can do that.

**Comment by christinem** [comment]
@david-yz-liu Changes made.

File: public/js/post/fill_textboxes.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ function fill300Textboxes(post, postElement) {
 
     for (var m = post.index300 + 1; m < activeCourses.length && post.filledTextboxes300 !== post.textboxes300; m++) {
         var course = activeCourses[m];
-        if (course.indexOf('CSC3') !== -1 && course.indexOf('CSC373') === -1 && 
-            course.indexOf('CSC369') === - 1) {
+        if (course.indexOf('CSC3') !== -1 && (post.name === 'major' ||  (post.name === 'specialist' &&
```
Comment (david-yz-liu) [inline public/js/post/fill_textboxes.js]: Spacing is off after `||`; don't use `=== true`

File: public/js/post/update_post.js [hunk 2]
```diff
@@ -251,3 +251,18 @@ function updateMatCreditCount() {
...
+
+/**
+ * Returns whether course is a specialist course or not
+ * @param {string} Name of course
+ * @return {boolean} True if course is a specialist, False otherwise
+**/
+function notSpecialistCourse(course) {
+    'use strict';
+
+    if (specialistCourses.indexOf(course) === -1) {
```
Comment (david-yz-liu) [inline public/js/post/update_post.js]: Just do `return specialistCourses.indexOf(course) === -1`

**Comment by christinem** [comment]
Sorry @david-yz-liu - I can make the requested above changes in a new branch since you merged this one.

**Comment by david-yz-liu** [comment]
@christinem I made the changes, don't worry.

**Pull request merged**